plone/app/registry/browser/records.py

```diff
@@ -58,16 +59,20 @@ def shouldPurge(self):
 
 
 def checkFieldName(val):
-    for letter in val:
-        if letter not in _valid_field_name_chars:
-            raise Invalid('Not a valid field name')
+    # Reuse same regex as in plone.registry.registry._Records to allow dottedname with one '/'
+    validkey = re.compile(
```

Comment:
Can you define this regex outside of the function? Then it should only be run once at import time, rather than every time the function is called.